src/iris/ui/static/js/iris.js

```diff
@@ -1438,11 +1444,46 @@ iris = {
     init: function(){
       var location = window.location.pathname.split('/'),
           path = this.data.id = location[location.length - 1];
+      Handlebars.registerPartial('comment', this.data.commentSource)
       this.getIncident(path);
     },
     events: function(){
       var data = this.data;
       data.$page.on('click', data.claimIncidentBtn, this.claimIncident.bind(this));
+      data.$page.on('click', data.showAddCommentBtn, this.showComment.bind(this));
+      data.$page.on('click', data.hideAddCommentBtn, this.hideComment.bind(this));
+      data.$page.on('click', data.addCommentBtn, this.addComment.bind(this))
+    },
+    showComment: function() {
+      $(this.data.addCommentContainer).show();
+      $(this.data.showAddCommentBtn).hide();
+      $('#comment-body').focus();
+    },
+    hideComment: function() {
+      $('#comment-body').val('');
+      $(this.data.addCommentContainer).hide();
+      $(this.data.showAddCommentBtn).show();
+    },
+    addComment: function() {
+      var self = this,
+          comment = {
+            author: window.appData.user,
+            content: $('#comment-body').val()
```

Comment:
Can we verify the given body isn't an empty string/a string with just whitespace before we allow the form to be submitted?